kong/plugins/basic-auth/access.lua

```diff
@@ -155,7 +155,13 @@ end
 
 
 local function fail_authentication()
-  return false, { status = 401, message = "Invalid authentication credentials" }
+  return false, {
+      status = 401,
+      message = "Invalid authentication credentials",
+      headers = {
+        ["WWW-Authenticate"] = realm
```

Comment:
We cannot remove `realm=""` part from the header as for the Basic auth it is required according to: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#basic